### PR TITLE
Tripal 4 Fields: Sequence Length + Sequence Checksum

### PR DESCRIPTION
**Pull request by pdtouch**
Code is cleaned up, removing unnecessary classes and code, added feature to not show checksum if sequence is not provided. Tested functionality

Issue #1414 

tv4g1-issue1414-data__sequence_length

## Instructions for Testing

1. Create an organism
2. Go to Tripal > Page Structure > Gene > Manage Fields
3. Delete the existing "Sequence" (bio_data_8_data_2044) and "Sequence Length" (bio_data_8_data_1249).
4. Add the following fields in this order:
    - Chado Sequence Residues
    - Chado Feature Sequence Length
    - Chado Feature Sequence Checksum
5. Then create a gene:
    - Go to Tripal > Content, click "add Tripal Content" and then "Gene"
    - Select an organism, enter an identifier, and then fill in "Sequence" in specific ways to test the field.

**How to add a field:**
On Tripal > Page Structure > Gene > Manage Fields

1. Click "Add field"
2. Select the field name from the "Add a new field" dropdown and enter a label. Click "Save and continue".
3. On the next form, select "feature" as the Chado Base Table and click "Save Field settings".
4. Then on the final page, leave the defaults and click "Save settings".

**Note: There is no widget for the sequence length and sequence checksum fields. Instead the value is calculated based on the sequence residues and then displayed by the formatter.**

### CODE REVIEW

**Comment by laceysanderson** [comment]
## Notes

- This PR is dependant upon #1604. It cannot be merged until that one is first.
- This PR includes the work of @srobb1 done in PR #1419. As such we are closing #1419 in favour of this one.
- Testing instructions for this PR are in #1419 
- We will want to ensure the new fields are automatically attached via the YAML.

Thanks @pdtouch <3

File: tripal_chado/src/Plugin/Field/FieldType/ChadoSequenceChecksumDefault.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+/**
+ * Plugin implementation of Default Tripal field for sequence data.
+ *
+ * @FieldType(
+ *   id = "chado_sequence_checksum_default",
+ *   label = @Translation("Chado Feature Sequence Checksum"),
+ *   description = @Translation("A chado feature sequence md5 checksum"),
+ *   default_widget = "chado_sequence_checksum_widget_default",
+ *   default_formatter = "chado_sequence_checksum_formatter_default"
+ * )
```
Comment (dsenalik) [inline tripal_chado/src/Plugin/Field/FieldType/ChadoSequenceChecksumDefault.php:18]: Add `*   cardinality = 1` here

**Review by dsenalik — CHANGES_REQUESTED** [review]
I followed the testing steps, and the field worked perfectly.
- [x] Entering a simple sequence


displays 

and I verified the correctness of this checksum
`echo -n cat | md5sum`
`d077f244def8a70e5ea758bd8352fcd8  -`

- [x] Changing the sequence updates the length and checksum correctly.

- [x] A long sequence is correctly wrapped to the same length as for Tripal 3.

- [x] If I add extra white space, it is removed when I save. I just checked, and this is what Tripal 3 does, so I guess this is correct.

- [x] I configured a default value, and it showed up correctly when creating a new gene

- [x] In my container I am seeing the field debugger output, but I never turned that on. I'm not sure yet if that's a problem with this PR or not. Checking it here, and reference on PR#1604 in this comment https://github.com/tripal/tripal/pull/1604#issuecomment-1682760409


Changes requested:
- [x] The cardinality for this field should be set permanently to 1. This field will only apply to the feature table, and you can only have one sequence. You can add
`*   cardinality = 1` I have a separate comment showing where

- [x] In ChadoSequenceWidgetDefault.php you use a variable `$rec_id_val` but it is never set in this function, something might be wrong here

- [x]  Figure out why testing is failing

**Comment by laceysanderson** [comment]
The testing is failing on this branch due to the inclusion of the code from PR #1604. It has been fixed in that PR so please update this branch as well.

**Comment by laceysanderson** [comment]
Tests are now passing!
Should just be the two remaining comments in @dsenalik review:

> The cardinality for this field should be set permanently to 1. This field will only apply to the feature table, and you can only have one sequence. You can add `cardinality = 1` I have a separate comment showing where.

> In ChadoSequenceWidgetDefault.php you use a variable $rec_id_val but it is never set in this function, something might be wrong here

**Review by dsenalik — APPROVED** [review]
Approved with suggested changes added.

**Pull request merged**